dtfs-central-api/src/services/state-machines/fee-record/fee-record.state-machine.test.ts

```diff
@@ -75,6 +77,47 @@ describe('FeeRecordStateMachine', () => {
       // Assert
       expect(handleFeeRecordPaymentAddedEvent).toHaveBeenCalledTimes(1);
     });
+
+    it(`handles the '${FEE_RECORD_EVENT_TYPE.CORRECTION_REQUESTED}' event`, async () => {
+      // Arrange
+      const stateMachine = FeeRecordStateMachine.forFeeRecord(TO_DO_FEE_RECORD);
+
+      // Act
+      await stateMachine.handleEvent({
+        type: FEE_RECORD_EVENT_TYPE.CORRECTION_REQUESTED,
+        payload: {
+          transactionEntityManager: {} as unknown as EntityManager,
+          requestedByUser: { id: '10000', firstName: 'Test', lastName: 'User' },
+          reasons: [RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT],
+          additionalInfo: 'some additional information',
+          requestSource: { platform: REQUEST_PLATFORM_TYPE.TFM, userId: 'abc123' },
+        },
+      });
+
+      // Assert
+      expect(handleFeeRecordCorrectionRequestedEvent).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe(`when the fee record has the '${FEE_RECORD_STATUS.PENDING_CORRECTION}' status`, () => {
+    // Arrange
+    const PENDING_CORRECTION_FEE_RECORD = FeeRecordEntityMockBuilder.forReport(UPLOADED_REPORT).withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION).build();
+
+    const VALID_PENDING_CORRECTION_FEE_RECORD_EVENT_TYPES: FeeRecordEventType[] = [];
+    const INVALID_PENDING_CORRECTION_FEE_RECORD_EVENT_TYPES = difference(FEE_RECORD_EVENT_TYPES, VALID_PENDING_CORRECTION_FEE_RECORD_EVENT_TYPES);
+
+    if (INVALID_PENDING_CORRECTION_FEE_RECORD_EVENT_TYPES.length !== 0) {
```

Comment:
I'm not sure, I just copied the existing pattern used in other blocks. That said I think I like it because it makes it clearer to the reader of the tests that there could well be nothing in the array and if so to not expect any tests of that kind to run.